test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetPackageSigningTests/RepositorySignedPackageTestCase.cs

```diff
@@ -0,0 +1,301 @@
+        {
+            // Arrange
+            EnsureVisualStudioHost();
+
+            var signedPackage = _fixture.RepositorySignedTestPackage;
+
+            using (var testContext = new ApexTestContext(VisualStudio, projectTemplate, XunitLogger))
+            {
+                await SimpleTestPackageUtility.CreatePackagesAsync(testContext.PackageSource, signedPackage);
+
+                var nugetConsole = GetConsole(testContext.Project);
+
+                nugetConsole.InstallPackageFromPMC(signedPackage.Id, signedPackage.Version);
+                testContext.Project.Build();
+                testContext.NuGetApexTestService.WaitForAutoRestore();
+
+                nugetConsole.UninstallPackageFromPMC(signedPackage.Id);
+                testContext.Project.Build();
+                testContext.NuGetApexTestService.WaitForAutoRestore();
+
+                CommonUtility.AssertPackageReferenceDoesNotExist(VisualStudio, testContext.Project, signedPackage.Id, signedPackage.Version, XunitLogger);
+            }
+        }
+
+        [CIOnlyNuGetWpfTheory]
+        [MemberData(nameof(GetPackagesConfigTemplates))]
+        public async Task UninstallFromPMCForPC_SucceedAsync(ProjectTemplate projectTemplate)
+        {
+            // Arrange
+            EnsureVisualStudioHost();
+
+            var signedPackage = _fixture.RepositorySignedTestPackage;
+
+            using (var testContext = new ApexTestContext(VisualStudio, projectTemplate, XunitLogger))
+            {
+                await SimpleTestPackageUtility.CreatePackagesAsync(testContext.PackageSource, signedPackage);
+
+                var nugetConsole = GetConsole(testContext.Project);
+
+                nugetConsole.InstallPackageFromPMC(signedPackage.Id, signedPackage.Version);
+                nugetConsole.UninstallPackageFromPMC(signedPackage.Id);
+
+                CommonUtility.AssertPackageNotInPackagesConfig(VisualStudio, testContext.Project, signedPackage.Id, signedPackage.Version, XunitLogger);
+            }
+        }
+
+        [CIOnlyNuGetWpfTheory]
+        [MemberData(nameof(GetPackageReferenceTemplates))]
+        public async Task UpdateUnsignedToSignedVersionFromPMCForPR_SucceedAsync(ProjectTemplate projectTemplate)
+        {
+            // Arrange
+            EnsureVisualStudioHost();
+
+            var packageVersion09 = "0.9.0";
+            var signedPackage = _fixture.RepositorySignedTestPackage;
+
+            using (var testContext = new ApexTestContext(VisualStudio, projectTemplate, XunitLogger))
+            {
+                await CommonUtility.CreatePackageInSourceAsync(testContext.PackageSource, signedPackage.Id, packageVersion09);
+                await SimpleTestPackageUtility.CreatePackagesAsync(testContext.PackageSource, signedPackage);
+
+                var nugetConsole = GetConsole(testContext.Project);
+
+                nugetConsole.InstallPackageFromPMC(signedPackage.Id, packageVersion09);
+                testContext.Project.Build();
```

Comment:
wouldn't build cause it to wait and also do  a restore?

Reply:
I should ideally, but as part of the ongoing apex stabilization `waitForAutoRestore` was added to ensure that after installing a package the assets file will be generated in the tests.